File: pizzayolo/src/main/java/pizzayolo/entity/Commande.java
package pizzayolo.entity;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.DiscriminatorType;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import org.hibernate.annotations.ManyToAny;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "commande")
@SequenceGenerator(name = "seqCommande", sequenceName = "seq_commande", initialValue = 1, allocationSize = 1)
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "type_commande", discriminatorType = DiscriminatorType.STRING, length = 5)

public abstract class Commande {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqCommande")
	@Column(name = "num_ticket")
	@JsonView(JsonViews.CommandeWithItem.class)
	protected Long numTicket;

	@JsonView(JsonViews.CommandeWithItem.class)
	@OneToMany(mappedBy = "idCB.commandeBoisson")
	Set<CommandeBoisson> boissons;

	@JsonView(JsonViews.CommandeWithItem.class)
	@OneToMany(mappedBy = "idCD.commandeDessert")
	Set<CommandeDessert> desserts;

	@JsonView(JsonViews.CommandeWithItem.class)
	@OneToMany(mappedBy = "commandePizza")
	Set<Pizza> pizzas;

	@ManyToOne
	@JoinColumn(name = "id_client", foreignKey = @ForeignKey(name = "COMMANDE_ID_CLIENT_FK"))
	@JsonView(JsonViews.CommandeWithItem.class)
	protected Utilisateur clientTicket;

	@Column(name = "date_ticket")
	@JsonView(JsonViews.CommandeWithItem.class)
	protected LocalDate dateTicket;

	@Column(name = "prix_total")
	@JsonView(JsonViews.CommandeWithItem.class)
	protected double prixTotal;

	public Commande() {
		this.dateTicket=LocalDate.now();
	}

	public Long getNumTicket() {
		return numTicket;
	}

	public void setNumTicket(Long numTicket) {
		this.numTicket = numTicket;
	}

	public Set<CommandeBoisson> getBoissons() {
		return boissons;
	}

	public void setBoissons(Set<CommandeBoisson> boissons) {
		this.boissons = boissons;
	}

	public Set<CommandeDessert> getDesserts() {
		return desserts;
	}

	public void setDesserts(Set<CommandeDessert> desserts) {
		this.desserts = desserts;
	}

	public Set<Pizza> getPizzas() {
		return pizzas;
	}

	public void setPizzas(Set<Pizza> pizzas) {
		this.pizzas = pizzas;
	}

	public Utilisateur getClientTicket() {
		return clientTicket;
	}

	public void setClientTicket(Utilisateur clientTicket) {
		this.clientTicket = clientTicket;
	}

	public LocalDate getDateTicket() {
		return dateTicket;
	}

	public void setDateTicket(LocalDate dateTicket) {
		this.dateTicket = dateTicket;
	}

	public double getPrixTotal() {
		return prixTotal;
	}

	public void setPrixTotal(double prixTotal) {
		this.prixTotal = prixTotal;
	}

	@Override
	public int hashCode() {
		return Objects.hash(numTicket);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Commande other = (Commande) obj;
		return Objects.equals(numTicket, other.numTicket);
	}

}

File: pizzayoloangular/src/app/model/commande.ts
import { Utilisateur } from './utilisateur';
import { Boisson } from './boisson';
import { Dessert } from './dessert';
import { Pizza } from './pizza';

export class Commande {
  public constructor(
    private _numTicket?: number,

    private _boissons?: Boisson[],
    private _desserts?: Dessert[],
    private _pizzas?: Pizza[],
    private _clientTicket?: Utilisateur,
    private _prixTotal?: number,
    private _typeCommande?: String,
    private _heureSouhaite?: String,
    private _place?: String
  ) {}

  get numTicket(): number | undefined {
    return this._numTicket;
  }

  set numTicket(value: number | undefined) {
    this._numTicket = value;
  }

  get boissons(): Array<Boisson> | undefined {
    return this._boissons;
  }

  set boissons(value: Array<Boisson> | undefined) {
    this._boissons = value;
  }

  get desserts(): Array<Dessert> | undefined {
    return this._desserts;
  }

  set desserts(value: Array<Dessert> | undefined) {
    this._desserts = value;
  }

  get pizzas(): Array<Pizza> | undefined {
    return this._pizzas;
  }

  set pizzas(value: Array<Pizza> | undefined) {
    this._pizzas = value;
  }

  get clientTicket(): Utilisateur | undefined {
    return this._clientTicket;
  }

  set clientTicket(value: Utilisateur | undefined) {
    this._clientTicket = value;
  }

  get prixTotal(): number | undefined {
    return this._prixTotal;
  }

  set prixTotal(value: number | undefined) {
    this._prixTotal = value;
  }

  get heureSouhaite(): String | undefined {
    return this._heureSouhaite;
  }

  set heureSouhaite(value: String | undefined) {
    this._heureSouhaite = value;
  }

  get place(): String | undefined {
    return this._place;
  }

  set place(value: String | undefined) {
    this._place = value;
  }

  get typeCommande(): String | undefined {
    return this._typeCommande;
  }

  set typeCommande(value: String | undefined) {
    this._typeCommande = value;
  }
}
